### PR TITLE
Add recommendation apis

### PR DESCRIPTION
**Pull request by qmhu**
#### What type of PR is this?

Feature

#### What this PR does / why we need it:

Add recommendation/recommendationRule api in craned server

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: pkg/server/router.go [hunk 1]
```diff
@@ -42,6 +40,27 @@ func (s *apiServer) installHandler() {
...
+			recommendv1.GET("", recommendationHandler.ListRecommendations)
+		}
+
+		// recommendationRules
+		recommendrulev1 := v1.Group("/recommendationRule")
+		{
+			recommendrulev1.GET("", recommendationHandler.ListRecommendationRules)
+			recommendrulev1.POST("", recommendationHandler.CreateRecommendationRule)
+			recommendrulev1.PUT("recommendationRuleName", recommendationHandler.UpdateRecommendationRule)
+			recommendrulev1.DELETE("recommendationRuleName", recommendationHandler.DeleteRecommendationRule)
```
Comment (zsnmwy) [inline pkg/server/router.go:56]:
```suggestion
			recommendrulev1.PUT(":recommendationRuleName", recommendationHandler.UpdateRecommendationRule)
			recommendrulev1.DELETE(":recommendationRuleName", recommendationHandler.DeleteRecommendationRule)
```

**Review by zsnmwy — CHANGES_REQUESTED** [review]

**Comment by zsnmwy** [comment]
https://www.postman.com/speeding-crater-505947/workspace/public/collection/14940923-8559deb0-9af9-4ace-bc64-14da3dd0c8f8?action=share&creator=14940923

File: pkg/server/handler/recommendation/recommendation.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+	ginwrapper.WriteResponse(c, nil, nil)
+
+	err := h.client.Delete(context.TODO(), recommendationRuleExist)
+	if err != nil {
+		ginwrapper.WriteResponse(c, err, nil)
+		return
+	}
+
+	ginwrapper.WriteResponse(c, nil, nil)
```
Comment (zsnmwy) [inline pkg/server/handler/recommendation/recommendation.go:112]: multi writeResponse

**Review by zsnmwy — CHANGES_REQUESTED** [review]

File: pkg/server/handler/recommendation/recommendation.go (hunk 2 above)
Comment (zsnmwy) [reply pkg/server/handler/recommendation/recommendation.go:112]: image

**Comment by qmhu** [comment]
@zsnmwy PTAL

**Review by zsnmwy — APPROVED** [review]

**Pull request merged**